packages/tx/src/eip4844Transaction.ts

```diff
@@ -0,0 +1,479 @@
+    const tx = decoded.message
+    const accessList: AccessListBuffer = []
+    for (const listItem of tx.accessList) {
+      const address = Buffer.from(listItem.address)
+      const storageKeys = listItem.storageKeys.map((key) => Buffer.from(key))
+      const accessListItem: AccessListBufferItem = [address, storageKeys]
+      accessList.push(accessListItem)
+    }
+    const to = tx.to.value === null ? undefined : Address.fromString(toHexString(tx.to.value))
+    const versionedHashes = tx.blobVersionedHashes.map((el) => Buffer.from(el))
+    const txData = {
+      ...tx,
+      ...{
+        versionedHashes,
+        to,
+        accessList,
+        r: decoded.signature.r,
+        s: decoded.signature.s,
+        v: BigInt(decoded.signature.yParity),
+        gasLimit: decoded.message.gas,
+      },
+    } as BlobEIP4844TxData
+    return new BlobEIP4844Transaction(txData, opts)
+  }
+
+  getUpfrontCost(baseFee: bigint = BigInt(0)): bigint {
+    const prio = this.maxPriorityFeePerGas
+    const maxBase = this.maxFeePerGas - baseFee
+    const inclusionFeePerGas = prio < maxBase ? prio : maxBase
+    const gasPrice = inclusionFeePerGas + baseFee
+    return this.gasLimit * gasPrice + this.value
+  }
+
+  raw(): TxValuesArray {
+    throw new Error('Method not implemented.')
+  }
+
+  /**
+   * Serialize a blob transaction to the execution payload variant
+   * @returns the minimum (execution payload) serialization of a signed transaction
+   */
+  serialize(): Buffer {
+    const to = {
+      selector: this.to !== undefined ? 1 : 0,
+      value: this.to?.toBuffer() ?? null,
+    }
+    const sszEncodedTx = SignedBlobTransactionType.serialize({
+      message: {
+        chainId: this.common.chainId(),
+        nonce: this.nonce,
+        maxPriorityFeePerGas: this.maxPriorityFeePerGas,
+        maxFeePerGas: this.maxFeePerGas,
+        gas: this.gasLimit,
+        to,
+        value: this.value,
+        data: this.data,
+        accessList: this.accessList.map((listItem) => {
+          return { address: listItem[0], storageKeys: listItem[1] }
+        }),
+        blobVersionedHashes: this.versionedHashes,
+        maxFeePerDataGas: this.maxFeePerDataGas,
+      },
+      // TODO: Decide how to serialize an unsigned transaction
+      signature: {
+        r: this.r ?? BigInt(0),
+        s: this.s ?? BigInt(0),
+        yParity: this.v === BigInt(1) ? true : false,
+      },
+    })
+    return Buffer.concat([TRANSACTION_TYPE_BUFFER, sszEncodedTx])
+  }
+
+  serializeNetworkWrapper(): Buffer {
+    const to = {
+      selector: this.to !== undefined ? 1 : 0,
+      value: this.to?.toBuffer() ?? null,
+    }
+    const blobArrays = this.blobs?.map((blob) => Uint8Array.from(blob)) ?? []
+    const serializedTxWrapper = BlobNetworkTransactionWrapper.serialize({
+      blobs: blobArrays,
+      blobKzgs: this.kzgCommitments?.map((commitment) => Uint8Array.from(commitment)) ?? [],
+      tx: { ...this.txData(), ...to },
```

Comment:
Is this really correct? With this spreading - regarding the structure of `txData()` - the values from `to` will appear twice in the data structure I guess?

Reply:
In this format, the properties in the object on the right should replace the equivalent properties in the object on the left, so the `to` property from the second will replace the first as in this [SO answer](https://stackoverflow.com/questions/49491393/using-spread-operator-to-update-an-object-value)

Reply:
Ah, interesting. 👍